stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/ChannelFilterRequest.kt

```diff
@@ -31,3 +30,26 @@ public object ChannelFilterRequest {
             )
         ).await()
 }
+
+/**
+ * Checks if the channel collection contains a channel, if yes then it returns skip handling result, otherwise it
+ * adds the channel.
+ */
+internal fun addIfChannelIsAbsent(channels: StateFlow<List<Channel>>, channel: Channel?): EventHandlingResult {
```

Comment:
Nit: For me these functions more suits `ChatEventHandler.kt` file. Or maybe this is the right time to create package `...querychannels.eventhandler`